pkg/server/bootstrap/identity.go

```diff
@@ -84,7 +84,23 @@ func NewConfigWithWildcardIdentities(config *rest.Config,
 	groupExportNames map[string]string,
 	grouptResourceExportNames map[schema.GroupResource]string,
 	kcpClient kcpclient.Interface) (identityConfig *rest.Config, resolve func(ctx context.Context) error) {
+	identityRoundTripper, identityResolver := NewWildcardIdentitiesRoundTripper(groupExportNames, grouptResourceExportNames, kcpClient)
+	identityConfig = rest.CopyConfig(config)
+	identityConfig.Wrap(identityRoundTripper)
+	return identityConfig, identityResolver
+}
 
+// NewWildcardIdentitiesRoundTripper creates a HTTP RoundTripper that injected resource identities for
+// individual group or group resources. Each group or resource is coming from one APIExport whose
+// names are passed in as a map.
+//
+// The returned resolve function will get the APIExports and extract the identities.
+// The resolve func might return an error if the APIExport is not found or for other reason. Only
+// after it succeeds a client using the returned RoundTripper can use the group and group resources
+// with identities.
+func NewWildcardIdentitiesRoundTripper(groupExportNames map[string]string,
```

Comment:
it's not really the roundtripper, but a roundtripper wrapper, right?

Reply:
it returns a RoundTripper and a function that allows for wrapping

Reply:
updated the function name and changed the gdoc.